src/libsodium/include/sodium/private/ed25519_ref10.h

```diff
@@ -99,6 +99,12 @@ void ge25519_double_scalarmult_vartime(ge25519_p2 *r, const unsigned char *a,
                                        const ge25519_p3 *A,
                                        const unsigned char *b);
 
+void
+ge25519_double_scalarmult_vartime_variable(ge25519_p2 *r, const unsigned char *a,
```

Comment:
I was thinking that instead of checking if the second input is the generator, we could have an additional argument, `unsigned int b`, that determines whether we are using the generator as a second parameter point or not. If yes, then that second ge25519_p3 would be any dummy variable with that type, as it would be ignored by the function. However, that might be quite confusing for readability. Maybe we could do this if we still expose both functions, but they internally use the same underlying function with this conditional argument.

Otherwise, we could include the generator as a constant, and force the caller of `double_scalarmult_vartime` to give the generator as an input. This, together with a new function `is_generator()` would also work. 

What direction you think is best?

Reply:
Couldn't `NULL` represent the generator?

Reply:
Oh, of course, yes, we could pass `NULL` as the second point.